flytekit/core/python_third_party_task.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+
+import os
+from typing import Any, Dict, List, Optional, Type, TypeVar
+
+from flyteidl.core import tasks_pb2 as _tasks_pb2
+
+from flytekit.common import utils as common_utils
+from flytekit.common.tasks.raw_container import _get_container_definition
+from flytekit.core.base_task import PythonTask, Task, TaskResolverMixin
+from flytekit.core.context_manager import FlyteContext, Image, ImageConfig, SerializationSettings
+from flytekit.core.resources import Resources, ResourceSpec
+from flytekit.core.shim_task import ExecutableTemplateShimTask, ShimTaskExecutor
+from flytekit.core.tracker import TrackedInstance
+from flytekit.loggers import logger
+from flytekit.models import task as _task_model
+from flytekit.models.core import identifier as identifier_models
+from flytekit.models.security import Secret, SecurityContext
+from flytekit.tools.module_loader import load_object_from_module
+
+TC = TypeVar("TC")
+
+
+class PythonThirdPartyContainerTask(ExecutableTemplateShimTask, PythonTask[TC]):
+    """
+    Please take a look at the comments for ``ExecutableTemplateShimTask`` as well. This class should be subclassed
```

Comment:
better documentation, that should explain the rationale behind this class? when to use it and why?

Reply:
ping?